app/controllers/reportings_controller.rb

```diff
@@ -197,18 +199,7 @@ def new
   end
 
   def create
-    @reporting = @project.reportings_via_source.build
-    @reporting.reporting_to_project_id = params['reporting']['reporting_to_project_id']
-
-    if @reporting.reporting_to_project.nil?
-      flash.now[:error] = l('timelines.reporting_could_not_be_saved')
-      render action: :new, status: :unprocessable_entity
-      return
-    end
-
-    check_visibility
-
-    if @reporting.save
+    if @reporting.reporting_to_project.present? && @reporting.project.visible? && @reporting.save
```

Comment:
First of all: @floriank did keep the original behaviour (except the 422 he introduced himself a day before), so from a refactoring standpoint everything is right here.

Nevertheless I have to point out, that the create action seems to be incomplete. There are only two vital checks performed before saving the reporting:

1. Is the reporting valid?
2. Is the user allowed to see the project where the reporting originates

Whats missing here is the question whether the user is **allowed to add reportings** at all.

Reading the [permissions.rb](https://github.com/opf/openproject/blob/9bb62ff5e4fa94e6866d4954e2f378ce33dfebf9/config/initializers/permissions.rb#L355-L358) I'd say it should be checked whether the user can `:edit_reportings`. However, strictly speaking this is no *refactoring* work.

@ulferts What do you think?